keylime/tenant.py

```diff
@@ -100,49 +100,50 @@ def __init__(self):
         self.registrar_ip = config.get('registrar', 'registrar_ip')
         self.webapp_ip = config.get('webapp', 'webapp_ip')
 
-        if config.getboolean('general', "enable_tls"):
-            self.context = self.get_tls_context()
+        if config.getboolean('general',"enable_tls"):
+            self.cloudverifier_context, self.registrar_context = self.get_tls_context()
         else:
-            logger.warning(
-                "TLS is currently disabled, keys will be sent in the clear! Should only be used for testing")
-            self.context = None
+            logger.warning("TLS is currently disabled, keys will be sent in the clear! Should only be used for testing")
+            self.cloudverifier_context = None
+            self.registrar_context = None
 
     def get_tls_context(self):
-        ca_cert = config.get('tenant', 'ca_cert')
-        my_cert = config.get('tenant', 'my_cert')
-        my_priv_key = config.get('tenant', 'private_key')
-        my_key_pw = config.get('tenant', 'private_key_pw')
-
-        tls_dir = config.get('tenant', 'tls_dir')
-
-        if tls_dir == 'default':
-            ca_cert = 'cacert.crt'
-            my_cert = 'client-cert.crt'
-            my_priv_key = 'client-private.pem'
-            tls_dir = 'cv_ca'
-
-        # this is relative path, convert to absolute in WORK_DIR
-        if tls_dir[0] != '/':
-            tls_dir = os.path.abspath('%s/%s' % (common.WORK_DIR, tls_dir))
-
-        if my_key_pw == 'default':
-            logger.warning(
-                "CAUTION: using default password for private key, please set private_key_pw to a strong password")
-
-        logger.info(f"Setting up client TLS in {tls_dir}")
-
-        ca_path = "%s/%s" % (tls_dir, ca_cert)
-        my_cert = "%s/%s" % (tls_dir, my_cert)
-        my_priv_key = "%s/%s" % (tls_dir, my_priv_key)
-
-        context = ssl.create_default_context()
-        context.load_verify_locations(cafile=ca_path)
-        context.load_cert_chain(
-            certfile=my_cert, keyfile=my_priv_key, password=my_key_pw)
-        context.verify_mode = ssl.CERT_REQUIRED
-        context.check_hostname = config.getboolean(
-            'general', 'tls_check_hostnames')
-        return context
+
+        contexts = []
+        for component in [ '', 'registrar_' ] :
+            ca_cert = config.get('tenant', component + 'ca_cert')
```

Comment:
This works ok during my tests so I think we are close to merging, one thing I wanted to check. Does this change require new values in `keylime.conf` ? I am trying to grok what `('tenant', component + 'my_cert')` is doing (mainly the component part?

Reply:
No new values required at all. Those are the certs that already exist on `keylime.conf`. Due to an unfortunate (seemingly) historical accident it looks like every verifier-related component (e.g., `my_cert`) does not have a prefix, while registrar-related components (e.g., `registrar_my_cert`) have a proper prefix. Please note that this is true for the sections `[cloud_verifier]` and `[registrar]`, but more importantly to our purposes, this is also true of the `[tenant]` section.